pkg/analysis/helpers/markers/analyzer.go

```diff
@@ -206,18 +296,40 @@ func NewMarkerSet(markers ...Marker) MarkerSet {
 	return ms
 }
 
-// Insert add the given markers to the MarkerSet.
+// Insert adds the given markers to the MarkerSet.
 // If any markers have the same value, the latter marker in the list
 // will take precedence, no duplication checks are implemented.
 func (ms MarkerSet) Insert(markers ...Marker) {
 	for _, marker := range markers {
-		ms[marker.Value] = marker
+		ms[marker.Identifier] = append(ms[marker.Identifier], marker)
 	}
 }
 
-// Has returns whether a marker with the value given is present in the
-// MarkerSet.
-func (ms MarkerSet) Has(value string) bool {
-	_, ok := ms[value]
+// Has returns whether marker(s) with the identifier given is present in the
+// MarkerSet. If Has returns true, there is at least one marker
+// with this identifier.
+func (ms MarkerSet) Has(identifier string) bool {
+	_, ok := ms[identifier]
 	return ok
 }
+
+func (ms MarkerSet) HasWithValue(marker string) bool {
```

Comment:
Ok this works quite nicely, you put the full marker and expression in and it will check for something matching, nice

Needs godoc, should explain how it is used in the godoc please

Reply:
https://github.com/JoelSpeed/kal/pull/7/commits/8156a607d1efbaa3cb3702e4c5b7e712925d482f